ls1intum/Artemis
# `General`: Sync progress bar color with theme

### Checklist
#### General


#### Client

### Motivation and Context



Fixes #5380.

### Description


- Subscribed to the theme service to automatically trigger updates
- Refactored the code so the template doesn't call functions anymore
- Added test for the component

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with a few exercises in different stages

Check that the progress bars still work and update with the theme automatically.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| progress-bar.component.ts | 100% | 100% |


### Screenshots



![ezgif-3-320252c4d9](https://user-images.githubusercontent.com/23171488/178814954-38cbc3cb-32b3-486b-bd90-3092685d4fa5.gif)
